### PR TITLE
trivial: Move the Fedora CI to a supported release

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/ci/Dockerfile-fedora.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM fedora:32
+FROM fedora:34
```
Comment (superm1) [inline contrib/ci/Dockerfile-fedora.in:1]: so this won't be affected until "tomorrow" if it's merged since we only build containers once a day.

Comment (hughsie) [reply contrib/ci/Dockerfile-fedora.in:1]: I'm cack-handedly trying to debug the Fedora CI issue of `meson.build:1:0: ERROR: Compiler cc can not compile programs` -- I actually think the `FROM fedora:34` is being ignored as `main.yml` is specifying `fwupd-${{matrix.os}}:latest` -- I wondered if the odd timing of the failure to coincide with the F34 release, but looking at https://hub.docker.com/_/fedora the `latest` is still listed as f33.

Comment (superm1) [reply contrib/ci/Dockerfile-fedora.in:1]: >I'm cack-handedly trying to debug the Fedora CI issue of meson.build:1:0: ERROR: Compiler cc can not compile programs -- I actually think the FROM fedora:34 is being ignored as main.yml is specifying fwupd-${{matrix.os}}:latest -- I wondered if the odd timing of the failure to coincide with the F34 release, but looking at https://hub.docker.com/_/fedora the latest is still listed as f33.

[build_diff.txt](https://github.com/fwupd/fwupd/files/6386266/build_diff.txt)

Here is the difference from the two container builds from day to day.   I don't see it pulling F34 in the diff.  But there are a bunch of packages that upgraded and presumably one of those caused the failure.

Comment (superm1) [reply contrib/ci/Dockerfile-fedora.in:1]: And FYI I got those from https://github.com/fwupd/fwupd/actions/workflows/create_containers.yml

Comment (hughsie) [reply contrib/ci/Dockerfile-fedora.in:1]: Interesting, good detective work! I guess the main ones of interest would be glibc, and rpm, the latter even more likely as it only seems to be happening in rpmbuild. I guess there's no way to test these as doing so would mean merging and waiting 24h for the CI to regenerate?

I tested this fedora:34 version with podman locally, and it worked, so I'm pondering merging this patch and seeing if the problem magically goes away in 24h? Insane?

Comment (superm1) [reply contrib/ci/Dockerfile-fedora.in:1]: I mean this seems like a real problem on fedora 32 no? Like anyone on Fedora 32 should be hitting it and able to reproduce it.

Comment (superm1) [reply contrib/ci/Dockerfile-fedora.in:1]: > I tested this fedora:34 version with podman locally, and it worked, so I'm pondering merging this patch and seeing if the problem magically goes away in 24h? Insane?

I wonder if there is a way to trigger container build on demand? That's really what we want.

As for moving to Fedora 34, I don't see a problem moving, 32 will eventually hit EOL anyways.

Comment (superm1) [reply contrib/ci/Dockerfile-fedora.in:1]: I think you can just re-run the build containers job here: https://github.com/fwupd/fwupd/runs/2443676441?check_suite_focus=true if someone from Fedora thinks it's been fixed in F32.

Comment (hughsie) [reply contrib/ci/Dockerfile-fedora.in:1]: I think F32 is kinda EOL now, so I don't really want to spent the time on tracking it down.

**Comment by superm1** [comment]
It seems whatever breakage happened in fedora 32 is now hitting Windows builds too, which is now pulling from Fedora 34 looking at the logs: https://app.circleci.com/pipelines/github/fwupd/fwupd/5906/workflows/89e6b4a3-3c5d-4f59-9dea-463d1c9acc05/jobs/22828

**Comment by hughsie** [comment]
Can we make one CI failure not cancel the others?

**Comment by superm1** [comment]
Yes but Usually it makes sense for the current behavior - it means you made a mistake that should be fixed.

in other words rather than burning 100 CI minutes you burn 10.

What exactly is going on with Fedora docker container? Seems like some wild bad problems if it's affecting compilers on both fedora 32 and 34.

**Comment by hughsie** [comment]
> in other words rather than burning 100 CI minutes you burn 10.

true.

> Seems like some wild bad problems if it's affecting compilers on both fedora 32 and 34.

I'm not exactly sure, but it seems to have been magically "fixed" by this morning. I do wonder if we should kill the various `$packagemanager update` commands from the various composes -- although then I guess it's not really representative anymore.

**Comment by superm1** [comment]
It looks fixed for fedora regular build but mingw still broken. Perhaps another dep missing still there on Fedora 34?

**Comment by superm1** [comment]
>I'm not exactly sure, but it seems to have been magically "fixed" by this morning. I do wonder if we should kill the various $packagemanager update commands from the various composes -- although then I guess it's not really representative anymore.

I would prefer to leave them, it's early indication of problems for stuff that _we_ depend upon.

**Pull request merged**